addons/auth_password_pwn_checker/models/res_users.py

```diff
@@ -0,0 +1,57 @@
+import hashlib
+import logging
+from datetime import timedelta, date
+
+import requests
+
+from odoo import _, models, api, fields
+
+_logger = logging.getLogger(__name__)
+
+
+class ResUsers(models.Model):
+    _inherit = ['res.users']
+
+    pwn_last_checked = fields.Date()
+
+    @classmethod
+    def authenticate(cls, db, login, password, user_agent_env):
+        uid = super(ResUsers, cls).authenticate(db, login, password, user_agent_env)
+        cls._check_user_password(uid, password)
+        return uid
+
+    @classmethod
+    def _check_user_password(cls, uid: int, password: str):
```

Comment:
Why not a real record-based method (just browse uid and call the method on it) ?